sdk/identity/Azure.Identity/src/Credentials/SharedTokenCacheCredential.cs

```diff
@@ -79,7 +79,8 @@ internal SharedTokenCacheCredential(string tenantId, string username, TokenCrede
                 tenantId,
                 sharedTokenCredentialOptions?.ClientId ?? Constants.DeveloperSignOnClientId,
                 null,
-                options ?? s_DefaultCacheOptions);
+                options ?? s_DefaultCacheOptions,
+                (options as SharedTokenCacheCredentialOptions)?.BeforeBuildClient);
```

Comment:
Now that we just pass options directly, could this extraction of the `BeforeBuildClient` value be done inside MsalPublicClient?

Reply:
I refactored this to expose extension through a new internal interface.